pkg/api/endpoints/util.go

```diff
@@ -34,42 +34,24 @@ import (
 // form for things like comparison.  The result is a newly allocated slice.
 func RepackSubsets(subsets []api.EndpointSubset) []api.EndpointSubset {
 	// First map each unique port definition to the sets of hosts that
-	// offer it.  The sets of hosts must be de-duped, using IP+UID as the key.
-	type addressKey struct {
-		ip  string
-		uid types.UID
-	}
+	// offer it.
 	allAddrs := map[addressKey]*api.EndpointAddress{}
 	portsToAddrs := map[api.EndpointPort]addressSet{}
 	for i := range subsets {
 		for j := range subsets[i].Ports {
-			epp := &subsets[i].Ports[j]
+			epp := subsets[i].Ports[j]
```

Comment:
Safe and wasteful?  It's a minor point, and guidance for Go code varies...

On Tue, Sep 15, 2015 at 6:00 PM, Clayton Coleman <notifications@github.com>
wrote:

> In pkg/api/endpoints/util.go
> <https://github.com/kubernetes/kubernetes/pull/13778#discussion_r39584404>
> :
>
> >  	allAddrs := map[addressKey]*api.EndpointAddress{}
> >  	portsToAddrs := map[api.EndpointPort]addressSet{}
> >  	for i := range subsets {
> >  		for j := range subsets[i].Ports {
> > -			epp := &subsets[i].Ports[j]
> > +			epp := subsets[i].Ports[j]
>
> It was not used in a pointer context and the copy by value should be safe
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13778/files#r39584404>.
>